Dockerfile

```diff
@@ -1,60 +1,83 @@
-# Use an official Python runtime based on Debian 10 "buster" as a parent image.
-FROM python:3.8.1-slim-buster
+# (Keep the version in sync with the node install below)
+FROM node:16 as frontend
 
-# Add user that will be used in the container.
-RUN useradd wagtail
+# Make build & post-install scripts behave as if we were in a CI environment (e.g. for logging verbosity purposes).
+ARG CI=true
 
-# Port used by this container to serve HTTP.
-EXPOSE 8000
+# Install front-end dependencies.
+COPY package.json yarn.lock webpack.config.js ./
+RUN yarn
+
+# Compile static files
+COPY ./apps/frontend/static_src/ ./apps/frontend/static_src/
+RUN yarn build
+
+
+# We use Debian images because they are considered more stable than the alpine
+# ones becase they use a different C compiler. Debian images also come with
+# all useful packages required for image manipulation out of the box. They
+# however weigh a lot, approx. up to 1.5GiB per built image.
+FROM python:3.9 as production
+
+# Install dependencies in a virtualenv
+ENV VIRTUAL_ENV=/venv
+
+RUN useradd guide --create-home && mkdir /app $VIRTUAL_ENV && chown -R guide /app $VIRTUAL_ENV
 
-# Set environment variables.
-# 1. Force Python stdout and stderr streams to be unbuffered.
-# 2. Set PORT variable that is used by Gunicorn. This should match "EXPOSE"
-#    command.
-ENV PYTHONUNBUFFERED=1 \
-    PORT=8000
-
-# Install system packages required by Wagtail and Django.
-RUN apt-get update --yes --quiet && apt-get install --yes --quiet --no-install-recommends \
```

Comment:
It might, but I'd want to be completely sure. We've been running the full-fat debian images in production for years without issue. Whilst it'd be nice to know whether this works 100% with what we need, I'd also want to test beforehand.

Although, perhaps this is the ideal testing ground :thinking: